services/wallet/transfer/transaction_manager_route.go

```diff
@@ -0,0 +1,346 @@
+	}
+
+	processorInputParams := pathprocessor.ProcessorInputParams{
+		FromAddr:  addressFrom,
+		ToAddr:    addressTo,
+		FromChain: path.FromChain,
+		ToChain:   path.ToChain,
+		FromToken: path.FromToken,
+		ToToken:   path.ToToken,
+		AmountIn:  path.AmountIn.ToInt(),
+		AmountOut: path.AmountOut.ToInt(),
+
+		Username:  username,
+		PublicKey: publicKey,
+		PackID:    packId,
+	}
+
+	data, err := pathProcessors[path.ProcessorName].PackTxInputData(processorInputParams)
+	if err != nil {
+		return types.Hash{}, err
+	}
+
+	addrTo := types.Address(addressTo)
+	sendArgs := &transactions.SendTxArgs{
+		Version: transactions.SendTxArgsVersion1,
+
+		// tx fields
+		From:                 types.Address(addressFrom),
+		To:                   &addrTo,
+		Value:                path.AmountIn,
+		Data:                 data,
+		Gas:                  (*hexutil.Uint64)(&path.TxGasAmount),
+		MaxFeePerGas:         path.MaxFeesPerGas,
+		MaxPriorityFeePerGas: path.TxPriorityFee,
+
+		// additional fields version 1
+		ValueOut:           path.AmountOut,
+		FromChainID:        path.FromChain.ChainID,
+		ToChainID:          path.ToChain.ChainID,
+		SlippagePercentage: slippagePercentage,
+	}
+	if path.FromToken != nil {
+		sendArgs.FromTokenID = path.FromToken.Symbol
+		sendArgs.ToContractAddress = types.Address(path.FromToken.Address)
+
+		// special handling for transfer tx if selected token is not ETH
+		// TODO: we should fix that in the trasactor, but till then, the best place to handle it is here
+		if !path.FromToken.IsNative() {
+			sendArgs.Value = (*hexutil.Big)(big.NewInt(0))
+
+			if path.ProcessorName == pathprocessor.ProcessorTransferName ||
+				path.ProcessorName == pathprocessor.ProcessorStickersBuyName ||
+				path.ProcessorName == pathprocessor.ProcessorENSRegisterName ||
+				path.ProcessorName == pathprocessor.ProcessorENSReleaseName ||
+				path.ProcessorName == pathprocessor.ProcessorENSPublicKeyName {
+				// TODO: update functions from `TransactorIface` to use `ToContractAddress` (as an address of the contract a transaction should be sent to)
+				// and `To` (as the destination address, recipient) of `SendTxArgs` struct appropriately
+				toContractAddr := types.Address(path.FromToken.Address)
+				sendArgs.To = &toContractAddr
+			}
+		}
+	}
+	if path.ToToken != nil {
+		sendArgs.ToTokenID = path.ToToken.Symbol
+	}
+
+	builtTx, usedNonce, err := pathProcessors[path.ProcessorName].BuildTransactionV2(sendArgs, lastUsedNonce)
+	if err != nil {
+		return types.Hash{}, err
+	}
+	txHash := signer.Hash(builtTx)
+	usedNonces[path.FromChain.ChainID] = int64(usedNonce)
+
+	tm.routerTransactions = append(tm.routerTransactions, &RouterTransactionDetails{
+		routerPath:   path,
+		txArgs:       sendArgs,
+		tx:           builtTx,
+		txHashToSign: types.Hash(txHash),
+	})
+
+	return types.Hash(txHash), nil
+}
+
+func (tm *TransactionManager) BuildTransactionsFromRoute(route routes.Route, addressFrom common.Address, addressTo common.Address,
```

Comment:
Same here for number of arguments
<img width="1082" alt="image" src="https://github.com/user-attachments/assets/565c2dfc-8177-4e63-99cd-e758952a65cf">

Reply:
Updated.